src/rootPages/Designer/ui_work_object_list_newObject_api.js

```diff
@@ -0,0 +1,245 @@
+   const UIClass = UI_Class(AB);
+   const L = UIClass.L();
+
+   class UI_Work_Object_List_NewObject_API extends UIClass {
+      constructor() {
+         super("ui_work_object_list_newObject_api", {
+            // component: base,
+
+            form: "",
+            buttonSave: "",
+            buttonCancel: "",
+         });
+
+         this.API_Read = UI_ApiObjectRead(AB);
+      }
+
+      ui() {
+         // Our webix UI definition:
+         return {
+            id: this.ids.component,
+            header: L("API"),
+            body: {
+               view: "form",
+               id: this.ids.form,
+               width: 800,
+               height: 500,
+               rules: {
+                  // TODO:
+                  // name: inputValidator.rules.validateObjectName
+               },
+               elements: [
+                  {
+                     rows: [
+                        {
+                           view: "text",
+                           label: L("Name"),
+                           name: "name",
+                           required: true,
+                           placeholder: L("Object name"),
+                           labelWidth: 70,
+                        },
+                        {
+                           view: "checkbox",
+                           label: L("Read Only"),
+                           value: "1",
+                           disabled: true,
+                        },
+                     ],
+                  },
+                  {
+                     view: "tabview",
+                     tabbar: {
+                        options: [
+                           { value: "Create", disabled: true },
+                           "Read",
+                           { value: "Update", disabled: true },
+                           { value: "Delete", disabled: true },
+                        ],
+                        value: "Read",
+                        height: 40,
+                     },
+                     cells: [this.API_Read.ui()],
+                  },
+                  {
+                     cols: [
+                        { fillspace: true },
+                        {
+                           view: "button",
+                           id: this.ids.buttonCancel,
+                           value: L("Cancel"),
+                           css: "ab-cancel-button",
+                           autowidth: true,
+                           click: () => {
+                              this.cancel();
+                           },
+                        },
+                        {
+                           view: "button",
+                           id: this.ids.buttonSave,
+                           css: "webix_primary",
+                           value: L("Save"),
+                           autowidth: true,
+                           type: "form",
+                           click: () => {
+                              return this.save();
```

Comment:
I'm wondering if we should disable the save until we have more functionality built? Or add some warning that it's not fully functional.

Reply:
Yes, We will not merge this merge until all of task series will be done.